extensions/schema-compare/src/schemaCompareResult.ts

```diff
@@ -0,0 +1,361 @@
+				this.generateScriptButton.enabled = true;
+			} else {
+				this.generateScriptButton.title = localize('schemaCompare.generateScriptButtonDisabledTitle', 'Generate script is enabled when the target is a database');
+			}
+		} else {
+			this.flexModel.addItem(this.noDifferencesLabel, { CSSStyles: { 'margin': 'auto' } });
+		}
+
+		let sourceText = '';
+		let targetText = '';
+		this.differencesTable.onRowSelected(() => {
+			let difference = this.comparisonResult.differences[this.differencesTable.selectedRows[0]];
+			if (difference !== undefined) {
+				sourceText = difference.sourceScript === null ? '\n' : this.getAggregatedScript(difference, true);
+				targetText = difference.targetScript === null ? '\n' : this.getAggregatedScript(difference, false);
+
+				this.diffEditor.updateProperties({
+					contentLeft: sourceText,
+					contentRight: targetText,
+					title: localize('schemaCompare.ObjectDefinitionsTitle', 'Object Definitions')
+				});
+			}
+		});
+	}
+
+	private getAllDifferences(differences: azdata.DiffEntry[]): string[][] {
+		let data = [];
+		if (differences) {
+			differences.forEach(difference => {
+				if (difference.differenceType === azdata.SchemaDifferenceType.Object) {
+					if (difference.sourceValue !== null || difference.targetValue !== null) {
+						data.push([difference.name, difference.sourceValue, this.SchemaCompareActionMap[difference.updateAction], difference.targetValue]);
+					}
+				}
+			});
+		}
+
+		return data;
+	}
+
+	private getAggregatedScript(diffEntry: azdata.DiffEntry, getSourceScript: boolean): string {
+		let script = '';
+		if (diffEntry !== null) {
+			script += getSourceScript ? diffEntry.sourceScript : diffEntry.targetScript;
+			diffEntry.children.forEach(child => {
+				let childScript = this.getAggregatedScript(child, getSourceScript);
+				if (childScript !== 'null') {
+					script += childScript;
+				}
+			});
+		}
+		return script;
+	}
+
+	private reExecute(): void {
+		this.flexModel.removeItem(this.splitView);
+		this.flexModel.removeItem(this.noDifferencesLabel);
+		this.flexModel.addItem(this.loader, { CSSStyles: { 'margin-top': '30px' } });
+		this.diffEditor.updateProperties({
+			contentLeft: '\n',
+			contentRight: '\n'
+		});
+		this.differencesTable.selectedRows = null;
+		this.resetButtons();
+		this.execute();
+	}
+
+	private createCompareButton(view: azdata.ModelView): void {
+		let runIcon = path.join(__dirname, '.', 'media', 'compare.svg');
+
+		this.compareButton = view.modelBuilder.button().withProperties({
+			label: localize('schemaCompare.compareButton', 'Compare'),
+			iconPath: runIcon,
+			title: localize('schemaCompare.compareButtonTitle', 'Compare')
+		}).component();
+
+		this.compareButton.onDidClick(async (click) => {
+			this.reExecute();
+		});
+	}
+
+	private createGenerateScriptButton(view: azdata.ModelView): void {
+		let fileIcon = path.join(__dirname, '.', 'media', 'generate-script.svg');
```

Comment:
is the '.' really needed here? I see we are using this for all the icon path

Reply:
wasn't needed. removed.

Reply:
Validated for Icon path without . on mac - works.